src/Swarm/TUI/View.hs

```diff
@@ -288,53 +291,98 @@ renderErrorDialog err = renderDialog (dialog (Just "Error") Nothing (maxModalWin
+  WinModal -> padBottom (Pad 1) $ hCenter $ txt "Congratulations!"
+  DescriptionModal e -> descriptionWidget s e
+  QuitModal -> padBottom (Pad 1) $ hCenter $ txt quitMsg
+  GoalModal g -> padLeftRight 1 (displayParagraphs g)
+
+quitMsg :: Text
+quitMsg = "Are you sure you want to quit this game and return to the menu?"
 
 -- | Generate a fresh modal window of the requested type.
 generateModal :: AppState -> ModalType -> Modal
-generateModal s mt = Modal mt (dialog (Just title) buttons (maxModalWindowWidth `min` requiredWidth)) widget
+generateModal s mt = Modal mt (dialog (Just title) buttons (maxModalWindowWidth `min` requiredWidth))
  where
   haltingMessage = case s ^. uiState . uiPrevMenu of
     NoMenu -> Just "Quit"
     _ -> Nothing
   descriptionWidth = 100
-  (title, widget, buttons, requiredWidth) =
+  (title, buttons, requiredWidth) =
     case mt of
-      HelpModal -> (" Help ", helpWidget, Nothing, maxModalWindowWidth)
-      RecipesModal ->
-        ( "Available Recipes"
-        , availableListWidget (s ^. gameState) RecipeList
-        , Nothing
-        , descriptionWidth
-        )
-      CommandsModal ->
-        ( "Available Commands"
-        , availableListWidget (s ^. gameState) CommandList
-        , Nothing
-        , descriptionWidth
-        )
+      HelpModal -> (" Help ", Nothing, maxModalWindowWidth)
+      RobotsModal -> ("Robots", Nothing, descriptionWidth)
+      RecipesModal -> ("Available Recipes", Nothing, descriptionWidth)
+      CommandsModal -> ("Available Commands", Nothing, descriptionWidth)
       WinModal ->
-        let winMsg = "Congratulations!"
-            continueMsg = "Keep playing"
+        let continueMsg = "Keep playing"
             stopMsg = fromMaybe "Pick the next game" haltingMessage
          in ( ""
-            , padBottom (Pad 1) $ hCenter $ txt winMsg
             , Just (0, [(stopMsg, Confirm), (continueMsg, Cancel)])
             , length continueMsg + length stopMsg + 32
             )
-      DescriptionModal e -> (descriptionTitle e, descriptionWidget s e, Nothing, descriptionWidth)
+      DescriptionModal e -> (descriptionTitle e, Nothing, descriptionWidth)
       QuitModal ->
-        let quitMsg = "Are you sure you want to quit this game and return to the menu?"
-            stopMsg = fromMaybe "Quit to menu" haltingMessage
+        let stopMsg = fromMaybe "Quit to menu" haltingMessage
          in ( ""
-            , padBottom (Pad 1) $ hCenter $ txt quitMsg
             , Just (0, [("Keep playing", Cancel), (stopMsg, Confirm)])
             , T.length quitMsg + 4
             )
-      GoalModal g -> (" Goal ", padLeftRight 1 (displayParagraphs g), Nothing, 80)
+      GoalModal _ -> (" Goal ", Nothing, 80)
+
+robotsListWidget :: GameState -> Widget Name
+robotsListWidget g = viewport RobotsViewport Vertical (hCenter table)
+ where
+  table =
+    BT.renderTable
+      . BT.setDefaultColAlignment BT.AlignCenter
```

Comment:
In ASCII art, I don't like the default of having lines drawn on all the borders.  It looks visually distracting and takes up a lot of space, especially vertical space.  How would it look to turn off the cell borders?  Can we do that + add enough space between things so that it looks OK?

Reply:
Agreed, in b2221fd the vertical border are removed, what do you think?

Reply:
That looks nice!